### PR TITLE
New fishing spots

### PR DESCRIPTION
**Pull request by cheekybrdy**
(A-Hydroponics) (C-Feature) (C-Sprites)
## About the PR 
Adds two new fishing spots and a new epic fish.

## Why's this needed? 
More content for fishing

```changelog
(u)Cheekybrdy
(+)Added 2 new fishing spots, the mainframe and the ice cream machine!
```

### CODE REVIEW

**Comment by frawhst** [comment]
You don’t have to close the PR every time there’s an error, you can just fix the code and push to the PR branch and it’ll update

**Comment by DisturbHerb** [comment]
For future reference, label macros can be found in `.github/keylabeler.yml` or in the default PR description template. They're generally something like [FEAT] or [QOL].

File: code/modules/fishing/fishing_spots.dm [hunk 1]
```diff
@@ -47,8 +47,7 @@ ABSTRACT_TYPE(/datum/fishing_spot)
 	user.visible_message("[user] [pick("reels in", "catches", "pulls in", "fishes up")] a \
 	[pick("big", "wriggly", "fat", "slimy", "fishy", "large", "high-quality", "nasty", "chompy", "real", "wily")] \
 	[prob(80) ? "[fish.name]" : pick("one", "catch", "chomper", "wriggler", "sunovagun", "sucker")]!")
-	user.put_in_hand_or_drop(fish)
-	//fish.set_loc(get_turf(user))
+	fish.set_loc(get_turf(user))
```
Comment (TobleroneSwordfish) [inline code/modules/fishing/fishing_spots.dm:50]: This seems like an unintentional change?

File: code/modules/fishing/fishing_spots.dm [hunk 2]
```diff
@@ -700,3 +699,21 @@ datum/fishing_spot/golden_toilet
...
+//ice cream machine
+/datum/fishing_spot/ice_cream_machine
+	fishing_atom_type = /obj/submachine/ice_cream_dispenser
+	rod_tier_required = 2
+	fish_available = list(/obj/item/reagent_container/food/snacks/condiment/chocchips = 40,\
+	/obj/item/reagent_container/food/snacks/condiment/custard = 30,\
+	/obj/item/reagent_container/food/snacks/condiment/cream = 20,\
+	/obj/item/reagent_container/food/snacks/condiment/syrup = 15,\
+	/obj/item/raw_material/ice = 35,\
+	/obj/item/reagent_contain/food/snacks/ice_cream/goodrandom)
```
Comment (TobleroneSwordfish) [inline code/modules/fishing/fishing_spots.dm:712]:
```suggestion
	/obj/item/reagent_contain/food/snacks/ice_cream/goodrandom = 30
	)
```
or something like that

File: code/modules/fishing/fishing_spots.dm [hunk 3]
```diff
@@ -700,3 +699,21 @@ datum/fishing_spot/golden_toilet
...
+
+
+//databank
+/datum/fishing_spot/databank
```
Comment (TobleroneSwordfish) [inline code/modules/fishing/fishing_spots.dm:716]: this is really stretching the concept of fishing lol, should probably have more than one fish at least

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]

File: code/modules/fishing/fishing_spots.dm (hunk 1 above)
Comment (Gannets) [reply code/modules/fishing/fishing_spots.dm]: Yeah this looks like it reverts back to the previous behaviour before I made fish attempt to get put in the player's open hand.

File: code/modules/fishing/fishing_spots.dm (hunk 3 above)
Comment (Gannets) [reply code/modules/fishing/fishing_spots.dm:718]: I don't mind a few silly ones, but yeah maybe this spot should include some random tapes/computer gear too?

File: code/modules/fishing/fish.dm [hunk 4]
```diff
@@ -373,6 +373,17 @@ Alien/mutant/other fish:
...
+//code
+/obj/item/fish/code_worm
+	name = "code worm"
+	desc = "This unstable creature has been swimming around in this code for a long time, giving developers and its victims a massive headache."
+	icon_state = "code_worm"
+	inhand_color = "#32CD32"
+	value  = FISH_RARITY_EPIC
+	New()
+		..()
+		name = "\improper [pick("free", "gamer", "Xtreme", "funny", "ultimate", "REAL")] [pick("cat", "puppy", "gaming", "fail", "cheat", "hax")] [pick("video", "content", "game", "images", "text", "audiobook", "podcast")].worm"
```
Comment (frawhst) [inline code/modules/fishing/fish.dm:385]: spaces in file names??? that's going to cause problems
```suggestion
		name = "\improper [pick("free", "gamer", "Xtreme", "funny", "ultimate", "REAL")]_[pick("cat", "puppy", "gaming", "fail", "cheat", "hax")] [pick("video", "content", "game", "images", "text", "audiobook", "podcast")].worm"
```

**Review by frawhst — COMMENTED** [review]
Changelog should be a bit more descriptive, would at least mention what the new spots are.

**Review by frawhst — APPROVED** [review]
lgtm

**Review by Gannets — APPROVED** [review]
👍

**Pull request merged**